### PR TITLE
Add Draw.actionable events

### PR DESCRIPTION
**Pull request by jisaacks**
This is in response to discussion on #520 

ping @mcwhittemore

### CODE REVIEW

File: src/modes/simple_select.js [hunk 1]
```diff
@@ -26,6 +27,32 @@ module.exports = function(ctx, options = {}) {
...
+      feature => feature instanceof MultiFeature
+    );
+
+    var combine = false;
+
+    if (selectedFeatures.length) {
+      var featureType = selectedFeatures[0].type.replace('Multi','');
+      selectedFeatures.forEach(feature => {
+        if(feature.type.replace('Multi','') === featureType) {
+          combine = true;
```
Comment (mcwhittemore) [inline src/modes/simple_select.js]: We should flip this logic to start as true and move to false if something that doesn't match the `featureType` is found.

File: src/lib/actionables.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const nextActionables = cb();
+    if (nextActionables.uncombine === prevActionables.uncombine &&
+        nextActionables.combine === prevActionables.combine &&
+        nextActionables.trash === prevActionables.trash) {
+          return; // Nothing changed
+    }
+    ctx.map.fire(Constants.events.ACTIONABLE, nextActionables);
+    Object.assign(prevActionables, nextActionables);
+  }
+}
```
Comment (mcwhittemore) [inline src/lib/actionables.js]: Hmmm... I bit worried about doing a noop check for some actionable changes and not for others. But I think this is ok for now. We can think a bit more about how to deduplicate these events a bit more later on.

Comment (jisaacks) [inline src/lib/actionables.js]: That makes sense, so should I remove the deduping effort here until there is a more broad solution?

**Comment by jisaacks** [comment]
@mcwhittemore Yep, I'll make the requested changes.

File: src/lib/actionables.js (hunk 2 above)
Comment (mcwhittemore) [inline src/lib/actionables.js]: +1

**Comment by jisaacks** [comment]
@mcwhittemore I made the requested change. Also let me know how those tests look. I am sure there could be more tests for several scenarios but I think that is a good start.

**Comment by mcwhittemore** [comment]
@jisaacks - this is looking good. Did you not want to add tests for the `direct_select` events as well?

Looks like this is conflicting with master. Can you rebase and I can do a final functional review and then we can merge this.

**Comment by jisaacks** [comment]
@mcwhittemore, sure, if I don't get to it today I'll do it tomorrow. Thanks

**Comment by jisaacks** [comment]
@mcwhittemore I added the requested tests and rebased. Thanks.

**Comment by mcwhittemore** [comment]
@jisaacks - Thanks. This is great. I changed the structure of the event a bit so future values won't get overwritten if they conflict with mapbox-gl-js values. I also changed `combine` to `combineFeatures` and `uncombine` to `uncombieFeatures` to match the Draw functions.

Thanks for this! I'm going to go incorporate it into a project now :)

This has been published as `mapbox-gl-draw@0.13.0` on npm.

**Comment by jisaacks** [comment]
Thanks @mcwhittemore, those changes you made make sense. I am going to be building something using this as well. If I encounter any issues I'll contribute back. :)

**Comment by mcwhittemore** [comment]
Sounds great. Thanks!

**Pull request merged**